lib/matplotlib/tests/test_usetex.py

```diff
@@ -80,6 +81,23 @@ def test_minus_no_descent(fontsize):
     assert len({*heights.values()}) == 1
 
 
+@pytest.mark.skipif(_has_tex_package('xcolor.sty'),
+                    reason='xcolor is not available')
+def test_usetex_xcolor():
+    mpl.rcParams['text.usetex'] = True
+
+    fig = plt.figure()
+    t = fig.text(0.5, 0.5, "Some text 0123456789")
+    fig.canvas.draw()
+    pos = t.get_window_extent()
+
+    mpl.rcParams['text.latex.preamble'] = r'\usepackage[dvipsnames]{xcolor}'
+    fig = plt.figure()
+    t = fig.text(0.5, 0.5, "Some text 0123456789")
+    fig.canvas.draw()
+    np.testing.assert_array_equal(t.get_window_extent(), pos)
```

Comment:
Minor nitpick: Can we make this more symmetrical? e.g.

```suggestion
    fig = plt.figure()
    text = fig.text(0.5, 0.5, "Some text 0123456789")
    fig.canvas.draw()

    mpl.rcParams['text.latex.preamble'] = r'\usepackage[dvipsnames]{xcolor}'
    fig = plt.figure()
    text2 = fig.text(0.5, 0.5, "Some text 0123456789")
    fig.canvas.draw()
    np.testing.assert_array_equal(text2.get_window_extent(),
                                  text.get_window_extent())
```